src/utils/version.rs

```diff
@@ -27,25 +27,19 @@ pub fn check_for_update(config: config::Config, force: bool) -> Result<()> {
     // check fs for last check time
     let last_checked_time = get_last_checked_time_from_disk(&version_file);
 
-    match last_checked_time {
-        Some(last_checked_time) => {
-            let time_since_check = current_time.duration_since(last_checked_time)?.as_secs();
-            tracing::debug!(
-                "Time since last update check: {:?}h",
-                time_since_check / ONE_HOUR
-            );
+    if is_foreground || last_checked_time.is_none() {
+        do_update_check(&mut checked, is_foreground)?;
+    } else if let Some(last_checked_time) = last_checked_time {
+        let time_since_check = current_time.duration_since(last_checked_time)?.as_secs();
+        tracing::debug!(
+            "Time since last update check: {:?}h",
+            time_since_check / ONE_HOUR
+        );
 
-            if force || time_since_check > ONE_DAY {
-                do_update_check(&mut checked)?;
-            } else {
-                tracing::debug!(
-                    "No need to check for updates. Automatic checks happen once per day"
-                );
-            }
-        }
-        // we haven't checked for updates before -- check now :)
-        None => {
-            do_update_check(&mut checked)?;
+        if time_since_check > ONE_DAY {
+            do_update_check(&mut checked, is_foreground)?;
+        } else {
+            tracing::debug!("No need to check for updates. Automatic checks happen once per day");
```

Comment:
i restructured this to try to make it a bit more readable - let me know if you think it has the opposite effect!